pytorch_lightning/plugins/precision/native_amp.py

```diff
@@ -109,9 +109,9 @@ def pre_optimizer_step(
 
     def autocast_context_manager(self) -> torch.cuda.amp.autocast:
         if self.use_cpu:
-            return torch.cpu.amp.autocast(fast_dtype=self._fast_dtype)
+            return torch.cpu.amp.autocast(dtype=self._fast_dtype)
```

Comment:
Can you also update the `self` attribute?

```python
pytorch_lightning/plugins/precision/native_amp.py:55:        self._fast_dtype = self._select_precision_dtype(precision)
pytorch_lightning/plugins/precision/native_amp.py:75:        return self._fast_dtype == torch.bfloat16
pytorch_lightning/plugins/precision/native_amp.py:112:            return torch.cpu.amp.autocast(fast_dtype=self._fast_dtype)
pytorch_lightning/plugins/precision/native_amp.py:114:            return torch.cuda.amp.autocast(fast_dtype=self._fast_dtype)
tests/plugins/test_amp_plugins.py:220:    assert context_manager.fast_dtype == torch.bfloat16
```

Reply:
Should we use the PyTorch version and use fast_dtype or dtype accordingly ? Would this work with work with lower version of PyTorch ?

Reply:
This will almost surely start breaking lower versions of pytorch. Then again bfloat16 will not work there... I guess this would break amp.autocast for `use_cpu` in general though for older versions of pytorch. Annoying.

Reply:
Actually when use_cpu is set, there is another check for pytorch 1.10 or higher, so that line would not fail for earlier versions since neither of those two amp.autocast lines should not be called unless either:

1) self.use_cpu is set
2) self.is_bfloat16 is set

use_cpu is enforced on pytorch 1.10+ in init:

```
        if use_cpu and not _TORCH_CPU_AMP_AVAILABLE:
            raise MisconfigurationException(
                "You have asked for native AMP on CPU, but AMP is only available on GPU for PyTorch 1.9 "
                "and lower. To use native AMP on CPU, install PyTorch 1.10 or later."
            )
```

and self.is_bfloat16 should only happen for pytorch 1.10+ as well. 

For other cases, it looks like the context manager doesn't have dtype (or previously fast_dtype) set at all, so it _should_ work.

So if this does create problems, it would be due to a bug and not a known issue.